### DISCUSSION TITLE
Water System in O3DE

### DISCUSSION THREAD

**Discussion by SuperABC** [o3de/o3de#14757] [General]
We have developed a water system in O3DE. And now, we'd like to contribute it to the community. Looking forward to your feedback, thanks.

**Implementation of features.**
1. New material type of water.
2. Water wave adjustment with 3 types of wave.
3. Color for water surface and underwater postprocessing.
4. Direct lighting on water surface.
5. Contact foam.
6. Dynamic normal textures.
7. Approximate subsurface scattering.
8. Transparency for large waves.
9. Dynamic surface mesh cascade.
10. Hybrid raytracing water component.
12. Screenspace water component.
13. Underwater postprocessing.
14. Surface bloom.

**Comment by ohmaya** [comment]
Hi SuperABC, do you have a branch? Thanks.

**Reply by SuperABC** [reply]
I'm now organizing the code to be submitted, the branch with water can be fetched about next week.

**Reply by ohmaya** [reply]
https://github.com/orgs/o3de/projects/44/views/1?filterQuery=water&pane=issue&itemId=21931861

**Reply by JailbreakPapa** [reply]
Amy update on the water branch being available?

**Reply by Puude** [reply]
Any updates? Thanks!

**Comment by Lykdraft** [comment]
Probably not. Peak contributions per week in 2021 were over **1500+**, 
2022 were around **~700**, 2023 dropped to **~250** average, beginning of 2024 dropped to **~160**, 
end of 2024 average **~40**, the **peak** in 2025, for now (23.5.2025) is **51**. (~**95%** decrease from 2021)

So, someone (Perhaps many?) left the ship.

**Comment by KylerSF** [comment]
Ok a lot to go over here.

O3DE contributors has definitely went down, but Amazon which was the biggest contributor as it started as their engine let go of a lot of developers or possibly moved some of them to other projects. So back in 2021 when their was a ton of contributors there was a lot that needed fixing and the engine was very unstable. Now there is still bugs, but the stability of a lot of the tooling has been addressed and is still being addressed when new bugs are discovered. 

The water system is created by Huawei, and at this time it most likely does not work. A lot has changed with the rendering pipeline of O3DE over the last couple years that it would likely need to be refactored for the newest version of o3de. A Huawei employee has been pushing to get some of these changes back into the main branch and not be in their internal branch, and they seem to have gotten the approval to push some of the changes back into the main o3de repo, I am unsure if the water system is one, but they are working on getting the Particle System up and running in the newest version of o3de. 

So the water system is probably on the back burner right now for Huawei, at least until the particle system is implemented (Which is the work of an employee on their free time from what I have read).

I hope this clears up any questions on the water system!

**Reply by Lykdraft** [reply]
>  but they are working on getting the Particle System up and running in the newest version of o3de.

Really? Obviously, that should be much higher on the TODO list than a water system. Indeed.
_(An internal 'Particle System' should be standard in any engine)_

Nonetheless, Huawei devs have not spoken about the water system for a long time.
I believe the last word was in 2023. 2 yrs. Long time, no?

**Reply by Lykdraft** [reply]
> The water system is created by Huawei, and at this time it most likely does not work. A lot has changed with the rendering pipeline of O3DE over the last couple years that it would likely need to be refactored for the newest version of o3de. A Huawei employee has been pushing to get some of these changes back into the main branch and not be in their internal branch, and they seem to have gotten the approval to push some of the changes back into the main o3de repo, ...
> 
> I hope this clears up any questions on the water system!

Well, your last github post went also unanswered in 2023. 
https://github.com/o3de/sig-content/issues/117#issuecomment-1741639134
How do you know that?

**Reply by guillaume-haerinck** [reply]
> > The water system is created by Huawei, and at this time it most likely does not work. A lot has changed with the rendering pipeline of O3DE over the last couple years that it would likely need to be refactored for the newest version of o3de. A Huawei employee has been pushing to get some of these changes back into the main branch and not be in their internal branch, and they seem to have gotten the approval to push some of the changes back into the main o3de repo, ...
> > I hope this clears up any questions on the water system!
> 
> Well, your last github post went also unanswered in 2023. [o3de/sig-content#117 (comment)](https://github.com/o3de/sig-content/issues/117#issuecomment-1741639134) How do you know that?

We get news from Joe Bryant who's in contact with them and also through the sig-graphics weekly meeting where Huawei engineer are usually part of. I actually saw this particle system in action during one of said meeting so its definitely coming.

**Reply by KylerSF** [reply]
Sorry for the late reply, like @guillaume-haerinck was saying most of my info is based on the thread in discord for the particle system, once in a while a Huawei employee talks about the water system as well. Truthfully the water system from them may never come, I feel from the posts its more on the back burner until the particle system is finished.

**Reply by KylerSF** [reply]
> > but they are working on getting the Particle System up and running in the newest version of o3de.
> 
> Really? Obviously, that should be much higher on the TODO list than a water system. Indeed. _(An internal 'Particle System' should be standard in any engine)_
> 
> Nonetheless, Huawei devs have not spoken about the water system for a long time. I believe the last word was in 2023. 2 yrs. Long time, no?

I totally agree! I wish there was more conversations on github where it is much easier to find the info on this.

**Reply by JailbreakPapa** [reply]
> > > The water system is created by Huawei, and at this time it most likely does not work. A lot has changed with the rendering pipeline of O3DE over the last couple years that it would likely need to be refactored for the newest version of o3de. A Huawei employee has been pushing to get some of these changes back into the main branch and not be in their internal branch, and they seem to have gotten the approval to push some of the changes back into the main o3de repo, ...
> > > I hope this clears up any questions on the water system!
> > 
> > 
> > Well, your last github post went also unanswered in 2023. [o3de/sig-content#117 (comment)](https://github.com/o3de/sig-content/issues/117#issuecomment-1741639134) How do you know that?
> 
> We get news from Joe Bryant who's in contact with them and also through the sig-graphics weekly meeting where Huawei engineer are usually part of. I actually saw this particle system in action during one of said meeting so its definitely coming.

did anyone catch a recording of the particle system in action?

**Reply by guillaume-haerinck** [reply]
I took this picture during the presentation 
![image](https://github.com/user-attachments/assets/ef489532-ebef-4409-a1ce-e2b242b7e822)

**Reply by JailbreakPapa** [reply]
> I took this picture during the presentation 
> 
> ![image](https://github.com/user-attachments/assets/ef489532-ebef-4409-a1ce-e2b242b7e822)
> 
> 

Thank you, Is it fine if I can share this internally? Some colleagues have also been looking for a status of the particle system, so this would be great to share with them.

**Reply by guillaume-haerinck** [reply]
No problem :)

**Reply by guillaume-haerinck** [reply]
The particle system is discussed here as well https://discord.com/channels/805939474655346758/1265005651822645309/1351366058233298944
